spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,274 @@
+        progress_layout.addWidget(self.cancel_button)
+        self._progress_widget.setLayout(progress_layout)
+
+        self._progress_label = QLabel(_('Downloading'))
+
+        self.install_info = QLabel(
+            _("Downloading the latest Spyder version <br>"))
+
+        button_layout = QHBoxLayout()
+        self.ok_button = QPushButton(_('OK'))
+        button_layout.addStretch()
+        button_layout.addWidget(self.ok_button)
+        button_layout.addStretch()
+        action_layout.addStretch()
+        action_layout.addWidget(self._progress_label)
+        action_layout.addWidget(self._progress_widget)
+        action_layout.addWidget(self.install_info)
+        action_layout.addSpacing(10)
+        action_layout.addLayout(button_layout)
+        action_layout.addStretch()
+
+        # Layout
+        general_layout = QHBoxLayout()
+        general_layout.addLayout(action_layout)
+
+        self.setLayout(general_layout)
+
+    def update_installation_status(self, status):
+        """Update installation status (downloading, installing, finished)."""
+        self._progress_label.setText(status)
+        self.install_info.setText(INSTALL_INFO_MESSAGES[status])
+        if status == INSTALLING:
+            self._progress_bar.setRange(0, 0)
+            self.cancel_button.hide()
+
+    def update_installation_progress(self, current_value, total):
+        """Update installation progress bar."""
+        self._progress_bar.setMaximum(total)
+        self._progress_bar.setValue(current_value)
+
+
+class UpdateInstallerDialog(QDialog):
+    """Update installer dialog."""
+
+    # Signal to get the download progress
+    # int: Download progress
+    # int: Total download size
+    sig_download_progress = Signal(int, int)
+
+    # Signal to get the current status of the update installation
+    # str: Status string
+    sig_installation_status = Signal(str)
+
+    def __init__(self, parent):
+
+        self.cancelled = False
+        self.status = NO_STATUS
+        self.thread_install_update = None
+        super().__init__(parent)
+        self.setWindowFlags(Qt.Dialog | Qt.MSWindowsFixedSizeDialogHint)
+        self._parent = parent
+        self._installation_widget = UpdateInstallation(self)
+
+        # Layout
+        installer_layout = QVBoxLayout()
+        installer_layout.addWidget(self._installation_widget)
+        self.setLayout(installer_layout)
+
+        # Signals
+        self.sig_download_progress.connect(
+            self._installation_widget.update_installation_progress)
+        self.sig_installation_status.connect(
+            self._installation_widget.update_installation_status)
+        self.sig_installation_status.connect(
+            self.finished_installation)
+
+        self._installation_widget.ok_button.clicked.connect(
+            self.close_installer)
+        self._installation_widget.cancel_button.clicked.connect(
+            self.cancel_install)
+
+        # Show integration widget
```

Comment:
I think this should be something like:
```suggestion
        # Show installation widget
```